ml_project/src/features/custom_transformer.py

```diff
@@ -0,0 +1,41 @@
+""" Make custom transformer """
+
+import logging
+from typing import NoReturn
+
+import pandas as pd
+from sklearn.preprocessing import (
+    StandardScaler,
+)
+
+from sklearn.base import (
+    BaseEstimator,
+    TransformerMixin,
+)
+
+# from ..entity.custom_transformer_params import TransformerParams
+
+
+logger = logging.getLogger(__name__)
+
+
+class CustomTransformer(BaseEstimator, TransformerMixin):
+    """ Custom transformer class """
+    def __init__(self, features) -> NoReturn:
+        """ Class ctor """
+        self.transform_numerical = StandardScaler()
+        self.numerical_features = features.numerical_features
+
+    def fit(
+        self,
+        data: pd.DataFrame,
+        ) -> pd.DataFrame:
+        """ Fitting transformer for numerical features"""
+        self.transform_numerical.fit(data[self.numerical_features])
+        return self
+
+    def transform(self,
+        data: pd.DataFrame,
```

Comment:
need to add validation ( fit or not, if not then raise exception )
![image](https://user-images.githubusercontent.com/49230518/168040698-3befce8d-fcd9-4d90-a8aa-f8b57a76654e.png)
![image](https://user-images.githubusercontent.com/49230518/168040795-061589e8-17ee-4bb7-83d3-48240d0b49e1.png)